### PR TITLE
[WIP] Optimize HelpersOperationBinaryAdd.c

### PR DESCRIPTION
**Pull request by riya-17**
### What does this PR do?
Optimize already existing code

### Why was it initiated? Any relevant Issues?
It is part of GSoC 2019 application as discussed with mentor [kayhayen](/kayhayen)

### PR Checklist
- [x] Created a function to reduce redundancy of if blocks.
- [x] Combined two if conditions into one.

### CODE REVIEW

**Comment by riya-17** [comment]
@kayhayen I can't find any existing issues concerning this project which you have assigned me as per the discussion on our video call. Please mention an issue in this PR for GSoC 2019.

**Comment by bratao** [comment]
This is a very basic commit and no optimization happens at all. I expect more from a GSoC candidate. Please try harder.

**Comment by riya-17** [comment]
@bratao this was more to understand what I am assigned to and what I have to do and to gather a clearer idea of the issue.

**Comment by kayhayen** [comment]
@riya-17 Please ignore that guy, he is apparently a troll.

File: nuitka/build/static_src/HelpersOperationBinaryAdd.c [hunk 1]
```diff
@@ -19,23 +19,27 @@
...
-        long b = PyInt_AS_LONG(operand2);
-
-        long i = (long)((unsigned long)a + b);
-
-        // Detect overflow, in which case, a "long" object would have to be
-        // created, which we won't handle here.
-        if ((i ^ a) >= 0 || (i ^ b) >= 0) {
-            return PyInt_FromLong(i);
-        }
+        long_var(PyObject *operand1, PyObject *operand2);
```
Comment (kayhayen) [inline nuitka/build/static_src/HelpersOperationBinaryAdd.c]: This looks like illegal C code.

File: nuitka/build/static_src/HelpersOperationBinaryAdd.c [hunk 2]
```diff
@@ -46,8 +50,7 @@ PyObject *BINARY_OPERATION_ADD_OBJECT_INT(PyObject *operand1, PyObject *operand2
 
         PyObject *x = slot1(operand1, operand2);
 
-        if (x != Py_NotImplemented) {
-            if (unlikely(x == NULL)) {
+        if (x != Py_NotImplemented && (unlikely(x == NULL)) {
```
Comment (kayhayen) [inline nuitka/build/static_src/HelpersOperationBinaryAdd.c]: This changes the meaning, the block contained more code, which is now not guarded.

File: nuitka/build/static_src/HelpersOperationBinaryAdd.c [hunk 3]
```diff
@@ -59,14 +62,7 @@ PyObject *BINARY_OPERATION_ADD_OBJECT_INT(PyObject *operand1, PyObject *operand2
...
     if (PyInt_Check(operand1)) {
-        long a = PyInt_AS_LONG(operand1);
-        long b = PyInt_AS_LONG(operand2);
-
-        long i = (long)((unsigned long)a + b);
-
-        if ((i ^ a) >= 0 || (i ^ b) >= 0) {
-            return PyInt_FromLong(i);
-        }
+        long_var(PyObject *operand1, PyObject *operand2);
```
Comment (kayhayen) [inline nuitka/build/static_src/HelpersOperationBinaryAdd.c]: See above, also when you do what I call outlining a function, don't forget to assign its return value, and consider that when you moved the return to its inside, it became no longer a return of this function, but the new one.

**Pull request closed without merging**